imap_processing/write_to_cdf.py

```diff
@@ -0,0 +1,97 @@
+import os
+
+import numpy as np
+import xarray as xr
+from cdflib.xarray import xarray_to_cdf
+
+
+def calc_start_time(shcoarse_time: int):
+    """Calculate the datetime64 from the CCSDS secondary header information.
+
+    Since all instrument has SHCOARSE or MET seconds, we need convert it to
+    UTC. Took this from IDEX code.
+
+    Parameters
+    ----------
+    shcoarse_time: int
+        Number of seconds since epoch (nominally the launch time)
+
+    Returns
+    -------
+    np.datetime64
+        The time of the event
+
+    TODO
+    -----
+    This conversion is temporary for now, and will need SPICE in the future.
+    Nick Dutton mentioned that s/c clock start epoch is
+        jan-1-2010-00:01:06.184 ET
+    We will use this for now.
+    """
+    # Get the datetime of Jan 1 2010 as the start date
+    launch_time = np.datetime64("2010-01-01T00:01:06.184")
+    return launch_time + np.timedelta64(shcoarse_time, "s")
+
+
+def write_to_cdf(
```

Comment:
This is already in cdf_utils. Can you combine this with that? Or leverage that one somehow?
https://github.com/IMAP-Science-Operations-Center/imap_processing/blob/10e75630bdca2637a8744356ffe3d873f7711d2b/imap_processing/cdf_utils.py#L46

Reply:
Since my upcoming PR is using that, I didn't want to change it. I created a issue to combine those two functions into one and updates CDF attrs to have filename parameters in the dataset as we discussed on Slack. I will do that in this issue: https://github.com/IMAP-Science-Operations-Center/imap_processing/pull/268